### PR TITLE
Playlist

### PR DESCRIPTION
**Pull request by DrabWeb**
Adds the visual playlist and changes `MusicController` to play through the playlist instead of random songs

### CODE REVIEW

**Comment by peppy** [comment]
No longer WIP after some large refactoring.

Deficiencies:
- Filtering doesn't work yet (relies on https://github.com/ppy/osu-framework/pull/569 i guess)
- Doesn't update when new beatmaps are added.

File: osu.Game/Overlays/Music/FilterControl.cs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        private void current_ValueChanged(string newValue) => FilterChanged?.Invoke(newValue);
+
+        public Action ExitRequested;
+
+        public Action<string> FilterChanged;
+
+        public class FilterTextBox : SearchTextBox
+        {
+            protected override Color4 BackgroundUnfocused => OsuColour.FromHex(@"222222");
+            protected override Color4 BackgroundFocused => OsuColour.FromHex(@"222222");
```
Comment (smoogipooo) [inline osu.Game/Overlays/Music/FilterControl.cs]: These should be coming from load().

File: osu.Game/Overlays/MusicController.cs [hunk 2]
```diff
@@ -81,229 +80,208 @@ protected override bool OnDragEnd(InputState state)
...
+                                                {
+                                                    Action = prev,
+                                                    Icon = FontAwesome.fa_step_backward,
+                                                },
+                                                playButton = new Button
+                                                {
+                                                    Scale = new Vector2(1.4f),
+                                                    IconScale = new Vector2(1.4f),
+                                                    Action = play
+                                                    ,
```
Comment (smoogipooo) [inline osu.Game/Overlays/MusicController.cs]: Lonely comma.

**Review by smoogipoo — CHANGES_REQUESTED** [review]
See comments. Then resolve changes + ready for merge.

**Pull request merged**